src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -388,16 +388,14 @@ void Simulator::handle_message_hil_sensor(const mavlink_message_t *msg)
 		float battery_percentage = 1.0f - (now_us - batt_sim_start) / discharge_interval_us;
 
 		battery_percentage = math::max(battery_percentage, _battery_min_percentage.get() / 100.f);
+
 		float vbatt = math::gradual(battery_percentage, 0.f, 1.f, _battery.empty_cell_voltage(), _battery.full_cell_voltage());
 		vbatt *= _battery.cell_count();
 
 		const float throttle = 0.0f; // simulate no throttle compensation to make the estimate predictable
-		_battery.updateBatteryStatus(now_us, vbatt, ibatt, true, true, 0, throttle, armed, &_battery_status);
-
+		_battery.updateBatteryStatus(now_us, vbatt, ibatt, true, true, 0, throttle, armed, true);
```

Comment:
Side note, this throttle value is kind of questionable. It's going to be wrong on VTOL.